packages/odata-service-writer/src/updates.ts

```diff
@@ -43,13 +44,15 @@ export function updateManifest(basePath: string, service: OdataService, fs: Edit
 export function updatePackageJson(path: string, fs: Editor, addMockServer: boolean) {
     const packageJson = JSON.parse(fs.read(path));
     packageJson.devDependencies = packageJson.devDependencies ?? {};
-    packageJson.ui5 = packageJson.ui5 ?? {};
-    packageJson.ui5.dependencies = packageJson.ui5.dependencies ?? [];
+    if (!hasUI5CliV3(packageJson.devDependencies)) {
```

Comment:
Im not sure about the logic here now, its a little difficult to follow as the presence of `pacakgeJson.ui5` is being used as a condition throughout but thats not really the thing thats relevant. I think this function should be refactored to be clearer and remove the conditions `packageJson.ui5` and replace with a more explicit condition.

```suggestion
   const ui5CliVersion = getVersion('@ui5/cli');
   ...
   // Add stuff common to all versions of @ui5/cli
   if (ui5CliVersion < '3') {
       // Add stuff for `ui5` section only
   } 
```

Reply:
I'll refactor this to add a block to make updates only when ui5/cli is lower than 3. Afaik the ui5 section is only for cli but not certain.